tests/test_templatetags.py

```diff
@@ -216,2484 +55,408 @@ class Media:
         """
-        rendered = template.render(Context({}))
         self.assertHTMLEqual(rendered, expected)
 
-    def test_default_slot_is_fillable_by_explicit_fill_content(self):
-        component.registry.register("test_comp", self.ComponentWithDefaultSlot)
-
-        template_str: types.django_html = """
+    @parametrize_context_behavior(["django", "isolated"])
+    def test_component_inside_block(self):
+        component.registry.register("slotted_component", SlottedComponent)
+        template: types.django_html = """
+            {% extends "block.html" %}
             {% load component_tags %}
-            {% component 'test_comp' %}
-              {% fill "main" %}<p>This fills the 'main' slot.</p>{% endfill %}
+            {% block body %}
+            {% component "slotted_component" %}
+                {% fill "header" %}{% endfill %}
+                {% fill "main" %}
+                TEST
+                {% endfill %}
+                {% fill "footer" %}{% endfill %}
             {% endcomponent %}
+            {% endblock %}
         """
-        template = Template(template_str)
+        rendered = Template(template).render(Context())
         expected = """
-            <div>
-            <main><p>This fills the 'main' slot.</p></main>
+            <!DOCTYPE html>
+            <html lang="en">
+            <body>
+            <main role="main">
+            <div class='container main-container'>
+                <custom-template>
+                <header></header>
+                <main>TEST</main>
+                <footer></footer>
+                </custom-template>
             </div>
+            </main>
+            </body>
+            </html>
         """
-        rendered = template.render(Context({}))
         self.assertHTMLEqual(rendered, expected)
 
-    def test_error_raised_when_default_and_required_slot_not_filled(self):
-        @component.register("test_comp")
-        class ComponentWithDefaultAndRequiredSlot(component.Component):
-            template: types.django_html = """
-                {% load component_tags %}
-                <div>
-                    <header>{% slot "header" %}Your Header Here{% endslot %}</header>
-                    <main>{% slot "main" default required %}Easy to override{% endslot %}</main>
-                </div>
-            """
-
-        template_str: types.django_html = """
-            {% load component_tags %}
-            {% component 'test_comp' %}
-            {% endcomponent %}
-        """
-        template = Template(template_str)
-        with self.assertRaises(TemplateSyntaxError):
-            template.render(Context({}))
-
-    def test_fill_tag_can_occur_within_component_nested_in_implicit_fill(
-        self,
-    ):
-        component.registry.register("test_comp", self.ComponentWithDefaultSlot)
-        component.registry.register("slotted", SlottedComponent)
+    # TODO: https://github.com/EmilStenstrom/django-components/issues/508
```

Comment:
I found out that there's an issue with using `block` tags with `slot` tags (e.g. `slot` tag inside `block` tag), for the "django" mode. Created an issue for this bug at https://github.com/EmilStenstrom/django-components/issues/508